calyx-opt/src/passes/compile_sync.rs

```diff
@@ -46,7 +46,8 @@ impl Named for CompileSync {
     }
 }
 
-fn count_barriers(
+/// put into the count set the barrier indices appearing in the thread
+pub(super) fn count_barriers(
```

Comment:
Not a huge fan of making functions visible across passes. Maybe we can create a folder called `sync` under passes and have the shared code in one file and two implementations exposed as two different passes?

Reply:
It turns out that I did not even share this function across these passes. Somehow I thought I was going to and made this function public. Sorry.